tests/unit/transform/date-test.js

```diff
@@ -39,28 +38,22 @@ test('#deserialize', function(assert) {
   assert.equal(transform.deserialize(undefined), null);
 });
 
-test('#deserialize with different offset formats', function(assert) {
+testInDebug('#deserialize with different offset formats', function(assert) {
```

Comment:
I think we should likely always test this, but only test the expect deprecation in debug.